src/Core/Vipr/FileWriter.cs

```diff
@@ -26,8 +31,37 @@ public static void Write(IEnumerable<TextFile> textFilesToWrite, string outputDi
                 if (!Directory.Exists(Path.GetDirectoryName(filePath)))
                     Directory.CreateDirectory(Path.GetDirectoryName(filePath));
 
-                File.WriteAllText(filePath, file.Contents);
+                fileTasks.Add(WriteToDisk(filePath, file.Contents));
             }
+            await Task.WhenAll(fileTasks);
+        }
+
+        /**
+         * Write the file to disk. If the file is locked for editing,
+         * sleep until available
+         */
+        public static async Task WriteToDisk(string filePath, string output)
+        {
+            for (int tries = 0; tries < 10; tries++)
+            {
+                StreamWriter sw = null;

```

Comment:
Since this is a static class re-using a StreamWriter <here> per file is not easy to do. If this method is called often, creating a new writer per write might have a huge performance hit. We might want to add some perf counters here if we are finding writes are taking too long to consume.